plugins/scaffolder/src/components/TemplatePage/TemplatePage.tsx

```diff
@@ -36,50 +34,59 @@ import { scaffolderApiRef } from '../../api';
 import { rootRouteRef } from '../../routes';
 import { MultistepJsonForm } from '../MultistepJsonForm';
 
-const useTemplate = (
-  templateName: string,
-  catalogApi: typeof catalogApiRef.T,
-) => {
-  const { value, loading, error } = useAsync(async () => {
-    const response = await catalogApi.getEntities({
-      filter: { kind: 'Template', 'metadata.name': templateName },
-    });
-    return response.items as TemplateEntityV1alpha1[];
-  }, [catalogApi, templateName]);
-  return { template: value?.[0], loading, error };
+const useTemplateParameterSchema = (templateName: string) => {
+  const scaffolderApi = useApi(scaffolderApiRef);
+  const { value, loading, error } = useAsync(
+    () =>
+      scaffolderApi.getTemplateParameterSchema({
+        name: templateName,
+        kind: 'template',
+        namespace: 'default',
+      }),
+    [scaffolderApi, templateName],
+  );
+  return { schema: value, loading, error };
 };
 
-const OWNER_REPO_SCHEMA = {
-  $schema: 'http://json-schema.org/draft-07/schema#' as const,
-  required: ['storePath', 'owner'],
-  properties: {
-    owner: {
-      type: 'string' as const,
-      title: 'Owner',
-      description: 'Who is going to own this component',
-    },
-    storePath: {
-      type: 'string' as const,
-      title: 'Store path',
-      description:
-        'A full URL to the repository that should be created. e.g https://github.com/backstage/new-repo',
-    },
-    access: {
-      type: 'string' as const,
-      title: 'Access',
-      description: 'Who should have access, in org/team or user format',
-    },
-  },
+const storePathValidator = (
+  formData: { storePath?: string },
+  errors: FormValidation,
+) => {
+  const { storePath } = formData;
+  if (!storePath) {
+    return errors;
+  }
+
+  try {
+    const parsedUrl = parseGitUrl(storePath);
+
+    if (!parsedUrl.resource || !parsedUrl.owner || !parsedUrl.name) {
+      if (parsedUrl.resource === 'dev.azure.com') {
+        errors.storePath.addError(
+          "The store path should be formatted like https://dev.azure.com/{org}/{project}/_git/{repo} for Azure URL's",
+        );
+      } else {
+        errors.storePath.addError(
+          'The store path should be a complete Git URL to the new repository location. For example: https://github.com/{owner}/{repo}',
+        );
+      }
+    }
+  } catch (ex) {
+    errors.storePath.addError(
+      `Failed validation of the store path with message ${ex.message}`,
```

Comment:
```suggestion
      `Failed validation of the store path, ${ex.message}`,
```

Reply:
btw these are a bit redundant aren't they, isn't this text that's gonna go immediately below the store path input, no need to repeat the words saying what thing that failed validation